src/color-swatch/color-swatch.js

```diff
@@ -29,236 +28,119 @@ const Self = class ColorSwatch extends HTMLElement {
-	#initialize ({force} = {}) {
-		if (!force && this.#initialized) {
-			return;
-		}
-
-		this.#initialized = true;
-
-		this.#dom.wrapper = this.shadowRoot.querySelector("#wrapper");
-		this.#dom.colorWrapper = this.shadowRoot.querySelector("[part=color-wrapper]");
-		this.#dom.input = this.querySelector("input");
-		this.#dom.slot = this.shadowRoot.querySelector("slot:not([name])");
+		this.static = !input;
 
-		this.#dom.slot.addEventListener("slotchange", evt => {
-			this.#render();
-		});
+		// This should eventually be a custom state
+		this._el.wrapper.classList.toggle("static", this.static);
 
-		if (this.#dom.input) {
+		if (input && input !== previousInput) {
 			importIncrementable ??= import("https://incrementable.verou.me/incrementable.mjs").then(m => m.default);
-			this.#dom.input.addEventListener("input", evt => {
-				this.#render(evt);
-			});
-		}
-
-		this.verbatim = this.hasAttribute("verbatim");
-
-		if (this.verbatim) {
-			// Cannot display gamut info without parsing the color
-			this.setAttribute("gamuts", "none");
-		}
 
-		this.gamuts = null;
-		if (!this.matches('[gamuts="none"]')) {
-			this.gamuts = this.getAttribute("gamuts") ?? "srgb, p3, rec2020: P3+, prophoto: PP";
-			this.#dom.gamutIndicator = document.createElement("gamut-badge");
-
-			Object.assign(this.#dom.gamutIndicator, {
-				gamuts: this.gamuts,
-				id: "gamut",
-				part: "gamut",
-				exportparts: "label: gamutlabel",
-			});
-
-			this.#dom.colorWrapper.appendChild(this.#dom.gamutIndicator);
-
-			this.#dom.gamutIndicator.addEventListener("gamutchange", evt => {
-				let gamut = this.#dom.gamutIndicator.gamut;
-				this.setAttribute("gamut", gamut);
-				this.dispatchEvent(new CustomEvent("gamutchange", {
-					detail: gamut,
-				}));
+			input.addEventListener("input", evt => {
+				this.value = evt.target.value;
 			});
 		}
+	}
 
-		if (this.hasAttribute("property")) {
-			this.property = this.getAttribute("property");
-			this.scope = this.getAttribute("scope") ?? ":root";
-			this.#dom.style = document.createElement("style");
-			document.head.appendChild(this.#dom.style);
+	get gamut () {
+		return this._el.gamutIndicator.gamut;
+	}
 
-			let varRef = `var(${this.property})`;
-			if (this.verbatim) {
-				this.style.setProperty("--color", varRef);
-				this.value ||= varRef;
-			}
-			else {
-				let scopeRoot = this.closest(this.scope);
+	propChangedCallback ({name, prop, detail: change}) {
+		let input = this._el.input;
 
-				// Is contained within scope root
-				if (scopeRoot) {
-					this.style.setProperty("--color", varRef);
+		if (name === "gamuts") {
+			if (this.gamuts === "none") {
+				this._el.gamutIndicator?.remove();
+				this._el.gamutIndicator = null;
+			}
+			else if (this.gamuts) {
+				if (!this._el.gamutIndicator) {
+					this._el.colorWrapper.insertAdjacentHTML("beforeend", `
+						<gamut-badge id="gamut" part="gamut" exportparts="label: gamutlabel" gamuts="${ this.gamuts }"></gamut-badge>
```

Comment:
```suggestion
						<gamut-badge id="gamut" part="gamut" exportparts="label: gamutlabel" gamuts="${ this.gamuts }" color="${ this.color }"></gamut-badge>
```

Reply:
🤦🏽‍♀️🤦🏽‍♀️🤦🏽‍♀️

And this, my friends, is why coding until late is a Bad Idea™